### PR TITLE
feat(context): add `Context::aspect_ratio`

### PR DESCRIPTION
**Pull request by suprohub**

### CODE REVIEW

File: src/context/mod.rs [hunk 1]
```diff
@@ -89,6 +89,19 @@ impl Context {
         self.read(|ctx| (ctx.graphics.buffer_width, ctx.graphics.buffer_height))
     }
 
+    /// Calculates an aspect ratio of the drawable part of the window
```
Comment (tversteeg) [inline src/context/mod.rs:92]: Comment structure for public facing API only returning something is:

```rust
/// Short description.
///
/// Optional extra information.
///
/// # Returns
///
/// - ..
```

File: src/context/mod.rs [hunk 2]
```diff
@@ -89,6 +89,19 @@ impl Context {
...
+    #[must_use]
+    pub fn aspect_ratio(&self) -> (u32, u32) {
```
Comment (tversteeg) [inline src/context/mod.rs:94]: Public API `size` types are `f32`.

File: src/lib.rs [hunk 3]
```diff
@@ -548,4 +549,5 @@ impl<G: Game> ApplicationHandler<Context> for State<G> {
             self.event_loop_proxy = None;
         }
     }
+
```
Comment (tversteeg) [inline src/lib.rs:552]:
```suggestion
```

File: src/context/mod.rs [hunk 4]
```diff
@@ -89,6 +89,19 @@ impl Context {
...
+        // Greatest common divisor algorithm
+        fn gcd(a: u32, b: u32) -> u32 {
+            if b == 0 { a } else { gcd(b, a % b) }
```
Comment (tversteeg) [inline src/context/mod.rs:97]: Is it necessary to use a recursive function to calculate the aspect ratio?

**Review by tversteeg — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: src/context/mod.rs (hunk 2 above)
Comment (suprohub) [reply src/context/mod.rs:94]: ok

File: src/context/mod.rs (hunk 1 above)
Comment (suprohub) [reply src/context/mod.rs:92]: ok

File: src/context/mod.rs (hunk 4 above)
Comment (suprohub) [reply src/context/mod.rs:97]: ok no

Comment (suprohub) [reply src/context/mod.rs:97]: https://docs.rs/num/latest/num/integer/fn.gcd.html#

**Pull request closed without merging**